src/root_finding/automatic_differentiation.jl

```diff
@@ -40,6 +40,7 @@ end
 
 sin(x::Jet) = Jet(sin(x.val), x.der*cos(x.val))
 cos(x::Jet) = Jet(cos(x.val), -x.der*sin(x.val))
+tan(x::Jet) = Jet(tan(x.val), x.der / cos(x.val)^2 )
```

Comment:
There is a `sec(x)` function, so this could be rewritten in a julian way as `Jet(tan(x.val), x.der * sec(x) ^2)`